tests/s25Main/UI/testImageButton.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (C) 2005 - 2022 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "controls/ctrlImageButton.h"
+#include "ogl/ITexture.h"
+#include "uiHelper/uiHelpers.hpp"
+#include <rttr/test/random.hpp>
+#include <turtle/mock.hpp>
+#include <boost/test/unit_test.hpp>
+
+using namespace rttr::test;
+
+namespace {
+MOCK_BASE_CLASS(TestTexture, ITexture)
+{
+public:
+    TestTexture(Extent size) : size_(size) {}
+    Position GetOrigin() const override { return Position::all(0); }
```

Comment:
As you use the origin and size in the code, the test should likely also set the origin and maybe have multiple images (or you reset the size and/or origin in the test and show how the draw call changes.